packages/dataviews/src/components/dataviews-filters/index.tsx

```diff
@@ -98,47 +100,74 @@ export function FilterVisibilityToggle( {
 	if ( filters.length === 0 ) {
 		return null;
 	}
+
+	const addFilterButtonProps = {
+		label: __( 'Add filter' ),
+		'aria-expanded': false,
+		isPressed: false,
+	};
+	const toggleFiltersButtonProps = {
+		label: _x( 'Filter', 'verb' ),
+		'aria-expanded': isShowingFilter,
+		isPressed: isShowingFilter,
+		onClick: () => {
+			if ( ! isShowingFilter ) {
+				setOpenedFilter( null );
+			}
+			setIsShowingFilter( ! isShowingFilter );
+		},
+	};
+	const buttonComponent = (
+		<Button
+			ref={ buttonRef }
+			className="dataviews-filters__visibility-toggle"
+			size="compact"
+			icon={ funnel }
+			{ ...( hasVisibleFilters
+				? toggleFiltersButtonProps
+				: addFilterButtonProps ) }
+		/>
+	);
 	if ( ! hasVisibleFilters ) {
 		return (
 			<AddFilterMenu
 				filters={ filters }
 				view={ view }
 				onChangeView={ onChangeViewWithFilterVisibility }
 				setOpenedFilter={ setOpenedFilter }
-				trigger={
-					<Button
-						className="dataviews-filters__visibility-toggle"
-						size="compact"
-						icon={ funnel }
-						label={ __( 'Add filter' ) }
-						isPressed={ false }
-						aria-expanded={ false }
-					/>
-				}
+				trigger={ buttonComponent }
 			/>
 		);
 	}
 	return (
-		<div className="dataviews-filters__container-visibility-toggle">
-			<Button
-				className="dataviews-filters__visibility-toggle"
-				size="compact"
-				icon={ funnel }
-				label={ _x( 'Filter', 'verb' ) }
-				onClick={ () => {
-					if ( ! isShowingFilter ) {
-						setOpenedFilter( null );
-					}
-					setIsShowingFilter( ! isShowingFilter );
-				} }
-				isPressed={ isShowingFilter }
-				aria-expanded={ isShowingFilter }
-			/>
+		<FilterVisibilityToggle buttonRef={ buttonRef }>
+			{ buttonComponent }
 			{ hasVisibleFilters && !! view.filters?.length && (
 				<span className="dataviews-filters-toggle__count">
 					{ view.filters?.length }
 				</span>
 			) }
+		</FilterVisibilityToggle>
+	);
+}
+
+function FilterVisibilityToggle( {
+	buttonRef,
+	children,
+}: {
+	buttonRef: React.RefObject< HTMLButtonElement >;
+	children: React.ReactNode;
+} ) {
+	// Focus the `add filter` button when unmounts.
+	useEffect(
+		() => () => {
+			buttonRef.current?.focus();
+		},
+		[ buttonRef ]
+	);
+	return (
+		<div className="dataviews-filters__container-visibility-toggle">
```

Comment:
Why is this wrapper not always there? It will cause the button to unmount whenever `hasVisibilityFilters` changes. At worst, we could leave an empty div without the class?

Reply:
What do you mean always there? Can you elaborate a bit?

Reply:
done